apiml-security-common/src/main/java/org/zowe/apiml/security/common/login/LoginFilter.java

```diff
@@ -170,31 +178,36 @@ private static LoginRequest mapBase64Credentials(String base64Credentials) {
         try {
             credentials = Base64.getDecoder().decode(base64Credentials);
             int index = ArrayUtils.indexOf(credentials, (byte) ':');
-            if (index > 0) {
-                byte[] password = null;
-                char[] passwordChars;
-                try {
-                    password = Arrays.copyOfRange(credentials, index + 1, credentials.length);
-                    passwordChars = new char[password.length];
-                    for (int i = 0; i < password.length; i++) {
-                        passwordChars[i] = (char) password[i];
-                    }
-                    return new LoginRequest(
-                        new String(Arrays.copyOfRange(credentials, 0, index), StandardCharsets.UTF_8),
-                        passwordChars
-                    );
-                } finally {
+            if (index < 0) {
+                throw new BadCredentialsException("Invalid basic authentication header");
+            }
+            byte[] password = null;
+            char[] passwordChars;
+            try {
+                String username = new String(Arrays.copyOfRange(credentials, 0, index), StandardCharsets.UTF_8);
+                password = Arrays.copyOfRange(credentials, index + 1, credentials.length);
+                passwordChars = new char[password.length];
+                for (int i = 0; i < password.length; i++) {
+                    passwordChars[i] = (char) password[i];
+                }
+
+                if (StringUtils.isBlank(username) || passwordChars.length == 0) {
```

Comment:
In case of an empty username the passwordChars is not cleaned.

We can check the state even beofre the cycle, like:
```java
if (index < 0) {
     throw new BadCredentialsException("Invalid basic authentication header");
}
if ((index == 0) || (index == credentials.lenght - 1)) {
    throw new AuthenticationCredentialsNotFoundException("Username or password not provided.");
}
```